enhance Object examples (#29288)
mdn/content: files/en-us/web/javascript/reference/global_objects/object/index.md

const o3 = new Object(null);
```

### Using Object() constructor to turn primitives into an Object of their respective type

You can use the {{jsxref("Object/Object", "Object()")}} constructor to create an object wrapper of a primitive value.

The following examples create variables `o1` and `o2` which are objects storing {{jsxref("Boolean")}} and {{jsxref("BigInt")}} values:

```js

> // equivalent to const o = new Boolean(true)

Reviewer: ```suggestion
// Equivalent to const o = new Boolean(true)
```

> // Equivalent to const o = new Boolean(true)

Author: ```suggestion
// Equivalent to const o1 = new Boolean(true)
```

a little thing is hiding before merging 😄

const o3 = new Object(null);
```

> ### Using Object() constructor to create a specific type of object

Reviewer: ```suggestion
### Using Object() constructor to create primitive wrappers
```

Author: ```suggestion
### Using Object() constructor to turn primitives into an Object of their respective type
``` 
How about this one .... the same description we've used in the constructor page

Reviewer: I'm good with that

> You can use the [`Object()`](/en-us/docs/Web/JavaScript/Reference/Global_Objects/Object/Object#return_value) constructor to create an Object of a specific type.

Reviewer: ```suggestion
You can use the {{jsxref("Object/Object", "Object()")}} constructor to create an object wrapper of a primitive value.
```

Author: Actually I intended pointing to the return value section to let the user quickly explore its cases, but yeah it makes more sense to make it to the constructor page as a whole.

> The following examples store {{jsxref("Boolean")}} objects in `o`:

Reviewer: ```suggestion
The following examples store {{jsxref("Boolean")}} and {{jsxref("BigInt")}} values in `o1` and `o2`:
```

Author: ```suggestion
The following examples store {{jsxref("Boolean")}} and {{jsxref("BigInt")}} objects in `o1` and `o2`:
```
I think better saying objects here since it matches what this section about

Reviewer: I see what this sentence means. You don't "store" objects in variables; you "point" variables to values. The values here are objects, which store primitive values.

```suggestion
The following examples create variables `o1` and `o2` which are objects storing {{jsxref("Boolean")}} and {{jsxref("BigInt")}} values:
```

> const o = new Object(true);

Reviewer: ```suggestion
const o1 = new Object(true);

// No equivalent because BigInt() cannot be used a constructor
const o2 = new Object(1n);
```

Author: ```suggestion
const o1 = new Object(true);

// No equivalent because BigInt() doesn't create BigInt objects either calling it as a constructor or as a regular function
const o2 = new Object(1n);```

Reviewer: You cannot "call `BigInt()` as a constructor"—it simply throws an error `new BigInt()`. `BigInt()`, just like `Number()` or `String()`, return primitives.

Author: Yeah, I edited the suggestion right before your comment cuz I noticed that people might be confused with that and think that they can use constructor but it won't create an object .... I think the edit is better, please, check it again ....... my point was it can't be used to create an object (becasue it either throws an error if you call it as a constructor or returns a primitive value if you call it as a regular function)

Reviewer: Sure, this looks good
